src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -2444,6 +2584,106 @@ window.qBittorrent.DynamicTable ??= (() => {
         prevReverseSort: null,
         fileTree: new window.qBittorrent.FileTree.FileTree(),
 
+        initialize: function() {
+            this.collapseState = new Map();
+        },
+
+        isCollapsed: function(id) {
+            return this.collapseState.get(id)?.collapsed ?? false;
+        },
+
+        expandNode: function(id) {
+            if (this.collapseState.has(id))
+                this.collapseState.get(id).collapsed = false;
+            this._updateNodeState(id, false);
+        },
+
+        collapseNode: function(id) {
+            if (this.collapseState.has(id))
+                this.collapseState.get(id).collapsed = true;
+            this._updateNodeState(id, true);
+        },
+
+        expandAllNodes: function() {
+            for (const [key, ] of this.collapseState)
+                this.expandNode(key);
+        },
+
+        collapseAllNodes: function(depth = 1) {
```

Comment:
The function is collapsing ALL nodes, why should it need a depth? It looks confusing.